projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -7,219 +7,272 @@ import { useSelect } from '@wordpress/data';
+						return;
+					}
+
+					return tag;
+				} )
+				.filter( Boolean ); // Remove undefined values
+		},
+		[ tags ]
 	);
 
-	// For now categories are combined with tags into the same object. I am not sure they should
-	const taxonomies = categoryObjects.concat( tagObjects );
+	useEffect( () => {
+		setIsLoadingCategories( loading );
+	}, [ loading ] );
 
-	// We want to wait until all category names are loaded. This will return empty string (aka loading state) until all objects are truthy.
-	if ( taxonomies.filter( obj => ! obj || ! obj.name ).length ) {
-		return '';
-	}
+	const createPrompt = () => {
+		const shorter_content = content.slice( -240 );
 
+		// If title is not added, we will only complete.
+		if ( ! currentPostTitle ) {
+			return shorter_content;
+		}
+
+		// Title, content and categories
+		if ( shorter_content && categoryNames.length ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, comma-seperated category names and the last 240 characters of content. */
+				__( "This is a post titled '%1$s', published in categories '%2$s':\n\n … %3$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				currentPostTitle,
+				categoryNames,
+				shorter_content
+			);
+		}
+
+		// No content, only title and categories
+		if ( categoryNames.length ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and comma-seperated category names. */
+				__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				currentPostTitle,
+				categoryNames
+			);
+		}
+
+		// Title and content
+		if ( shorter_content ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and the last 240 characters of content. */
+				__( "This is a post titled '%1$s':\n\n…%2$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				currentPostTitle,
+				shorter_content
+			);
+		}
+
+		return sprintf(
+			/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
+			__( 'Write content of a post titled "%s"', 'jetpack' ),
+			currentPostTitle
+		);
+	};
+
+	const taxonomies = categoryObjects.filter( cat => cat.id !== 1 ).concat( tagObjects );
 	const categoryNames = taxonomies.map( ( { name } ) => name ).join( ', ' );
+	const contentBefore = useSelect( select => {
+		const editor = select( 'core/block-editor' );
+		const index = editor.getBlockIndex( clientId );
+		return editor.getBlocks().slice( 0, index ) ?? [];
+	} );
 
-	return createPrompt( currentPost.title, editorContent, categoryNames );
-}
+	const containsAiUntriggeredParapgraph = () => {
+		return (
+			contentBefore.filter(
+				block => block.name && block.name === 'jetpack/ai-paragraph' && ! block.attributes.content
```

Comment:
Could we somehow get  this from index.js?

Reply:
I don't think it would get dynamically updated if we add this logic in `index.js`, as if we change the blocks order from the UI...

Reply:
I mean `jetpack/ai-paragraph` should be stored there, but this is fine as well.

Reply:
Oh! You meant shoud NOT be stored there. Now I get it. I'll modify it.

Reply:
Done in 3b60c88524d4c22f75d2ac72f1b0460f9be5ca08